frontends/web/src/components/transactions/components/details.tsx

```diff
@@ -0,0 +1,187 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ITransaction, IAmount, getTransaction } from '@/api/account';
+import { A } from '@/components/anchor/anchor';
+import { Dialog } from '@/components/dialog/dialog';
+import { FiatConversion } from '@/components/rates/rates';
+import { Amount } from '@/components/amount/amount';
+import { Note } from '@/components/transactions/note';
+import { TxDetail } from './detail';
+import { Arrow } from './arrow';
+import { TxDateDetail } from './date';
+import { TxStatusDetail } from './status';
+import { TxAddressOrTxID } from './address-or-txid';
+import parentStyle from '@/components/transactions/transaction.module.css';
+
+type TProps = {
+  open: boolean;
+  onClose: () => void;
+  accountCode: string;
+  internalID: string;
+  note: string;
+  status: ITransaction['status'];
+  type: ITransaction['type'];
+  numConfirmations: number;
+  numConfirmationsComplete: number;
+  time: string | null;
+  amount: IAmount;
+  sign: string;
+  typeClassName: string;
+  feeRatePerKb: IAmount;
+  explorerURL: string;
+}
+
+export const TxDetailsDialog = ({
+  open,
+  onClose,
+  accountCode,
+  internalID,
+  note,
+  status,
+  type,
+  numConfirmations,
+  numConfirmationsComplete,
+  time,
+  amount,
+  sign,
+  typeClassName,
+  feeRatePerKb,
+  explorerURL,
+}: TProps) => {
+  const { t } = useTranslation();
+
+  const [transactionInfo, setTransactionInfo] = useState<ITransaction | null>();
```

Comment:
`transactionInfo` here has the (implicit) type of `ITransaction | null | undefined`. 

nit: Should we just do the following:

```suggestion
  const [transactionInfo, setTransactionInfo] = useState<ITransaction | null>(null);
```

? 🤔 

Since `getTransaction` returns either `ITransaction | null` anyway (not `undefined`), at least from its type.

Reply:
yes definitely